django/db/backends/sqlite3/operations.py

```diff
@@ -370,8 +375,27 @@ def subtract_temporals(self, internal_type, lhs, rhs):
             return 'django_time_diff(%s, %s)' % (lhs_sql, rhs_sql), params
         return 'django_timestamp_diff(%s, %s)' % (lhs_sql, rhs_sql), params
 
-    def insert_statement(self, ignore_conflicts=False):
-        return 'INSERT OR IGNORE INTO' if ignore_conflicts else super().insert_statement(ignore_conflicts)
+    def insert_statement(self, on_conflicts=ON_CONFLICTS_NONE):
+        if on_conflicts == ON_CONFLICTS_IGNORE:
+            return 'INSERT OR IGNORE INTO'
+
+        elif on_conflicts == ON_CONFLICTS_UPDATE and Database.sqlite_version_info < (3, 24, 0):
+            return 'INSERT OR REPLACE INTO'
+
+        return super().insert_statement(on_conflicts=on_conflicts)
```

Comment:
```suggestion
    def insert_statement(self, on_conflict=None):
        if on_conflict == OnConflict.IGNORE:
            return 'INSERT OR IGNORE INTO'
        if on_conflict == OnConflict.UPDATE and Database.sqlite_version_info < (3, 24, 0):
            return 'INSERT OR REPLACE INTO'
        return super().insert_statement(on_conflict)
```

Please also update to use a feature flag as mentioned in https://github.com/django/django/pull/13065#discussion_r668440287.